src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -730,6 +781,11 @@ namespace AppInstaller::CLI::Workflow
             }
         }
 
+        if (shouldReboot)
+        {
+           InitiateRebootIfApplicable(context, true);

```

Comment:
Changed to setting a context flag so that indicates whether we should initiate a reboot so that it only takes place at the end of all the install workflows and is contained inside a single workflow function.